client/payment-request/index.js

```diff
@@ -638,4 +646,55 @@ jQuery( ( $ ) => {
 		};
 		testImg.src = background;
 	}
+
+	/**
+	 * Get a list of enabled payment request methods.
+	 */
+	async function getEnabledPaymentRequestMethods() {
+		if ( getEnabledPaymentRequestMethodsCache ) {
+			return getEnabledPaymentRequestMethodsCache;
+		}
+
+		const getEnabledMethods = async function ( paymentRequest ) {
+			const result = await paymentRequest.canMakePayment();
+			let ret = [];
+
+			if ( ! result ) {
+				return ret;
+			}
+
+			if ( result.applePay ) {
+				ret = [ 'applePay' ];
+			} else if ( result.googlePay ) {
+				ret = [ 'googlePay' ];
+			}
+
+			getEnabledPaymentRequestMethodsCache = ret;
+			return ret;
+		};
+
+		if ( wcpayPaymentRequestParams.is_product_page ) {
+			const paymentRequest = getPaymentRequest( {
+				stripe: api.getStripe(),
+				total: wcpayPaymentRequestParams.product.total.amount,
+				requestShipping:
+					wcpayPaymentRequestParams.product.needs_shipping,
+				displayItems: wcpayPaymentRequestParams.product.displayItems,
+			} );
+
+			return getEnabledMethods( paymentRequest );
+		}
+
+		const cart = await api.paymentRequestGetCartDetails();
+		const paymentRequest = getPaymentRequest( {
+			stripe: api.getStripe(),
+			total: cart.total.amount,
+			requestShipping: cart.needs_shipping,
+			displayItems: cart.displayItems,
+		} );
+
+		return getEnabledMethods( paymentRequest );
+	}
+
+	window.getEnabledPaymentRequestMethods = getEnabledPaymentRequestMethods;
```

Comment:
I'm not sure whether a jQuery `ready` function is the best place for declaring the global method. I could see issues where a third party consumer tries to use the method before `window.getEnabledPaymentRequestMethods` is defined. However this probably won't affect your use-case with WooCommerce Analytics.

Perhaps, could a [filter](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-hooks/) be used to pass this data up to WooCommerce Analytics?

Reply:
Thank you @adrianduffell. I think we need to make sure to document it if we decide to go with global method approach, but the filter sounds good 👍 I'll look into it.

Reply:
@adrianduffell Updated code to use `wp.hooks` in https://github.com/Automattic/woocommerce-payments/pull/2610/commits/e796f8e95de9b3fbd6e395c6031cd524e049706c